clang/test/Driver/sycl-host-compiler.cpp

```diff
@@ -0,0 +1,55 @@
+// Tests the abilities involved with using an external host compiler
+// REQUIRES: clang-driver
+
+/// enabling with -fsycl-host-compiler
+// RUN: %clangxx -fsycl -fsycl-host-compiler=/some/dir/g++ %s -### 2>&1 \
+// RUN:  | FileCheck -check-prefix=HOST_COMPILER %s
+// HOST_COMPILER: clang{{.*}} "-fsycl-is-device"{{.*}} "-fsycl-int-header=[[INTHEADER:.+\.h]]"
+// HOST_COMPILER: g++{{.*}} "-I" "{{.*}}bin{{[/\\]+}}..{{[/\\]+}}include{{[/\\]+}}sycl"{{.*}} "-o" "[[HOSTOBJ:.+\.o]]"{{.*}} "-include" "[[INTHEADER]]"
+// HOST_COMPILER: ld{{.*}} "[[HOSTOBJ]]"
+
+// RUN: %clang_cl -fsycl -fsycl-host-compiler=/some/dir/cl %s -### 2>&1 \
+// RUN:  | FileCheck -check-prefix=HOST_COMPILER_CL %s
+// HOST_COMPILER_CL: clang{{.*}} "-fsycl-is-device"{{.*}} "-fsycl-int-header=[[INTHEADER:.+\.h]]"
+// HOST_COMPILER_CL: cl{{.*}} "-Fo[[HOSTOBJ:.+\.obj]]"{{.*}} "-I" "{{.*}}bin{{[/\\]+}}..{{[/\\]+}}include{{[/\\]+}}sycl"{{.*}} "-FI" "[[INTHEADER]]"
+// HOST_COMPILER_CL: link{{.*}} "[[HOSTOBJ]]"
+
+/// check for additional host options
+// RUN: %clangxx -fsycl -fsycl-host-compiler=g++ -fsycl-host-compiler-options="-DFOO -DBAR" %s -### 2>&1 \
+// RUN:  | FileCheck -check-prefix=HOST_OPTIONS %s
+// HOST_OPTIONS: g++{{.*}} "-o" "[[HOSTOBJ:.+\.o]]"{{.*}} "-DFOO" "-DBAR"
```

Comment:
I believe an MSVC version would also be nice to have, given the different parsing styles of the two compilers.